matchtree.go

```diff
@@ -779,6 +785,7 @@ func (d *indexData) newMatchTree(q query.Q) (matchTree, error) {
 
 		return &symbolRegexpMatchTree{
 			regexp:    regexp,
+			all:       regexp.String() == "(?i)(?-s:.)*", // .*
```

Comment:
can you add a test case that this condition actually works? Right now I can imagine us regressing on this and not noticing since only perf would be affected, not correctness.